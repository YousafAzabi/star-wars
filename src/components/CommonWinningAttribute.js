import React from 'react';

import InputLabel from '@material-ui/core/InputLabel';
import MenuItem from '@material-ui/core/MenuItem';
import FormControl from '@material-ui/core/FormControl';
import Select from '@material-ui/core/Select';
import { withStyles } from '@material-ui/core/styles';

import config from '../config';

const styles = theme => ({
  formControl: {
    float: 'left',
    width: '34%',
    minWidth: 200,
    marginLeft: '8%',
    marginRight: '8%'
  },
});

class TypeSelect extends React.Component {

  render() {
    const { classes, type, winningAttribute, handleChange } = this.props;
    const attributes = config.ATTRIBUTES[type];

    return (
      <FormControl className={classes.formControl}>
        <InputLabel htmlFor="demo-controlled-open-select">Winning Attribute</InputLabel>
        <Select value={winningAttribute}
        onChange={handleChange}>
          { attributes && attributes.map(attribute =>
            <MenuItem key={attribute} value={attribute}>{attribute}</MenuItem>
          )}
        </Select>
      </FormControl>
    );
  }
}

export default withStyles(styles)(TypeSelect);
