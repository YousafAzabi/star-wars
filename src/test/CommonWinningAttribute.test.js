import React from "react";
import { create } from "react-test-renderer";
import CommonWinningAttribute from "../components/CommonWinningAttribute";

describe("CommonWinningAttribute component", () => {
  test("it matches the snapshot", () => {
    const component = create(<CommonWinningAttribute
      type='people'
      winningAttribute='mass'
      handleChange={console.log('change')} />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  test("it matches the snapshot", () => {
    const component = create(<CommonWinningAttribute
      type='people'
      winningAttribute='anything'
      handleChange={console.log('change')} />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  test("test display value equals 'mass'", () => {
    const component = create(<CommonWinningAttribute
      type='people'
      winningAttribute='mass'
      handleChange={console.log('change')} />
    );
    const rootInstance = component.root;
    const select = rootInstance.findByType('input');
    expect(select.props.value).toBe('mass');
  });

});
