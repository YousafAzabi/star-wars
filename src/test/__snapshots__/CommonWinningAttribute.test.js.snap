// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonWinningAttribute component it matches the snapshot 1`] = `
<div
  className="MuiFormControl-root TypeSelect-formControl-1"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
    data-shrink={true}
    htmlFor="demo-controlled-open-select"
  >
    Winning Attribute
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <div
      aria-haspopup="true"
      aria-pressed="false"
      className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      mass
    </div>
    <input
      type="hidden"
      value="mass"
    />
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root MuiSelect-icon"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;

exports[`CommonWinningAttribute component it matches the snapshot 2`] = `
<div
  className="MuiFormControl-root TypeSelect-formControl-1"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
    data-shrink={true}
    htmlFor="demo-controlled-open-select"
  >
    Winning Attribute
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <div
      aria-haspopup="true"
      aria-pressed="false"
      className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input MuiInputBase-inputSelect"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "&#8203;",
          }
        }
      />
    </div>
    <input
      type="hidden"
      value="anything"
    />
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root MuiSelect-icon"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</div>
`;
